surfactant/cmd/cli.py

```diff
@@ -17,3 +62,96 @@ def edit(sbom):
 @click.command("add")
 def add(sbom):
     "CLI command to add specific entry(s) to a supplied SBOM"
+
+
+class cli_find:
+    """
+    A class that implements the surfactant cli find functionality
+
+    Attributes:
+    match_functions     A dictionary of functions that provide matching functionality for given SBOM fields (i.e. uuid, sha256, installpath, etc)
+    sbom                A internal record of sbom entries the class adds to as it finds more matches.
+    """
+
+    match_functions: dict
+    camel_case_conversions: dict
+    sbom: SBOM
+
+    def __init__(self):
+        """Initializes the cli_find class"""
+        self.match_functions = {
+            int: self.match_single_value,
+            str: self.match_single_value,
+            list: self.match_array_value,
+        }
+        self.camel_case_conversions = {
+            "uuid": "UUID",
+            "filename": "fileName",
+            "containerpath": "containerPath",
+            "installpath": "installPath",
+            "capturetime": "captureTime",
+            "relationshipassertion": "relationshipAssertion",
+        }
+        self.sbom = SBOM()
+
+    def execute(self, input_sbom: SBOM, **kwargs):
+        """Executes the main functionality of the cli_find class
+        param: input_sbom   The sbom to find matches within
+        param: kwargs:      Dictionary of key/value pairs indicating what features to match on
+        """
+        for sw in input_sbom.software:
+            match = True
+            for k, v in kwargs.items():
+                if k == "file":
+                    entry_value = sw.sha256
+                    v, _, _ = self._calculate_hashes(v, sha256=True)
```

Comment:
This can be a time consuming operation, especially for large files (and repeated for each of possibly thousands of software entries). Can this be changed so that the hash only gets calculated once? Maybe the camel case conversion can also be done so that it doesn't need to happen multiple times.

Reply:
Ah yea good point! Updated to only perform hash calculation once at the beginning